src/LayerPlan.cpp

```diff
@@ -1407,6 +1456,68 @@ void LayerPlan::addWall(
         // Second pass to add the scarf closure
         addSplitWallPass(true);
     }
+}
+
+void LayerPlan::addWall(
+    const ExtrusionLine& wall,
+    size_t start_idx,
+    const Settings& settings,
+    const GCodePathConfig& default_config,
+    const GCodePathConfig& roofing_config,
+    const GCodePathConfig& bridge_config,
+    coord_t wall_0_wipe_dist,
+    const double flow_ratio,
+    bool always_retract,
+    const bool is_closed,
+    const bool is_reversed,
+    const bool is_linked_path,
+    const bool scarf_seam,
+    const bool smooth_speed)
+{
+    if (wall.empty())
+    {
+        return;
+    }
+
+    double non_bridge_line_volume = max_non_bridge_line_volume; // assume extruder is fully pressurised before first non-bridge line is output
+    const coord_t min_bridge_line_len = settings.get<coord_t>("bridge_wall_min_length");
+
+    addWallWithScarfSeam(
+        PathAdapter(wall),
+        start_idx,
+        settings,
+        default_config,
+        flow_ratio,
+        always_retract,
+        is_closed,
+        is_reversed,
+        wall.inset_idx_ == 0,
+        scarf_seam,
+        smooth_speed,
+        [this, &settings, &default_config, &roofing_config, &bridge_config, &non_bridge_line_volume](
```

Comment:
At some point it may be better (or at least more clear) to just to specify `[&]` instead ('everything by reference'), especially if you already need to include `this`.

Reply:
That's right, I know the use of [&] is somehow discouraged so I tend to never use it, but in this case it does make sense.

Reply:
I think the main reason for that is because it can pull in things like `this` unexpectedly. (And all the globals, etc.) However, in this case you're doing most of that anyway :-)